src/main/java/net/openhft/chronicle/core/internal/threads/VanillaThreadLock.java

```diff
@@ -0,0 +1,165 @@
+package net.openhft.chronicle.core.internal.threads;
+
+import net.openhft.chronicle.core.Jvm;
+import net.openhft.chronicle.core.OS;
+import net.openhft.chronicle.core.threads.InterruptedRuntimeException;
+import net.openhft.chronicle.core.threads.ThreadLock;
+import net.openhft.chronicle.core.values.LongValue;
+
+import java.io.File;
+
+public class VanillaThreadLock implements ThreadLock {
+    static final Metrics METRICS = new Metrics();
+    private static final long TOP_BITS = ~0L << 32;
+    private static final int UNLOCKED = 0;
+    private final LongValue twoThreadId;
+    private final long timeoutMs;
+    int busyLoopCount = 20_000;
+    int busyLockSlowerCount = 100;
+    private long oldLock;
+
+    public VanillaThreadLock(LongValue twoThreadId, long timeoutMs) {
+        this.twoThreadId = twoThreadId;
+        this.timeoutMs = timeoutMs;
+    }
+
+    @Override
+    public boolean tryLock(int threadId) throws IllegalStateException {
```

Comment:
I feel like we should call `Posix.getId()` in here rather than taking the thread id as a parameter or otherwise we'll forever be taking support questions why it doesn't work when people wrongly pass `Thread.currentThread().getId()` in.

Reply:
We should _at least_ rename the parameter to `posixThreadId` and describe where to get it from in the Javadoc

Reply:
Maybe even check the value and warn if they pass in a value equal to `Thread.currentThread().getId()`, although that could be noisy in the event that value just happened to be equal to the posix thread ID.

Reply:
Could double-check, however warnings can be easily ignored by developers.

Reply:
I have added some convience methods to make it clearer how it should be used.